src/application/application-roles/application-roles.service.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  BadRequestException,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { randomUUID } from 'crypto';
+import { RoleDto, UpdateRoleDto } from 'src/dto/application.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+
+@Injectable()
+export class ApplicationRolesService {
+  private readonly logger: Logger;
+  constructor(private readonly prismaService: PrismaService) {
+    this.logger = new Logger();
+  }
+
+  async createRole(data: RoleDto, applicationsId: string, roleId?: string) {
+    const application = await this.prismaService.application.findUnique({
+      where: { id: applicationsId },
+    });
+    if (!application) {
+      throw new HttpException(
+        'Application with provided application id dont exists',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+    const { description, name, isDefault, isSuperRole } = data;
+    const id = data.id ? data.id : roleId ? roleId : randomUUID();
+    try {
+      const newRole = await this.prismaService.applicationRole.create({
+        data: {
+          id,
+          description,
+          name,
+          isDefault,
+          isSuperRole,
+          applicationsId,
+        },
+      });
+      this.logger.log('New role added!', newRole);
+      return {
+        message: 'successfully created a new role',
+        role: newRole,
+        applicationsId,
+      };
+    } catch (error) {
+      this.logger.log('Error creating a new Role', error);
+      return {
+        message: 'error creating the role',
+        role: data,
+      };
+    }
+  }
+
+  async getRole() {}
```

Comment:
Throw not yet implemented error

Reply:
Please elaborate on this review.

Reply:
This is a blank function. If its not implement we should not leave dangling functions even if its not being called. If there is a scope of implementing in future we can throw Not yet implemented or some sort of error that states this function is not yet implemented.